<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.45">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Julien Roux, University of Basel, SIB">
<meta name="dcterms.date" content="2025-09-18">

<title>Hands-on: studying the long-lasting effects of Tamoxifen treatment on the CNS transcriptome across sexes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="hands_on_limma_files/libs/clipboard/clipboard.min.js"></script>
<script src="hands_on_limma_files/libs/quarto-html/quarto.js"></script>
<script src="hands_on_limma_files/libs/quarto-html/popper.min.js"></script>
<script src="hands_on_limma_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="hands_on_limma_files/libs/quarto-html/anchor.min.js"></script>
<link href="hands_on_limma_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="hands_on_limma_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="hands_on_limma_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="hands_on_limma_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="hands_on_limma_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Hands-on: studying the long-lasting effects of Tamoxifen treatment on the CNS transcriptome across sexes</h1>
            <p class="subtitle lead">SIB course: Best Practices for Investigating Gene Expression Differences by Biological Sex</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Julien Roux, University of Basel, SIB </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 18, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#loading-data-in-r" id="toc-loading-data-in-r" class="nav-link" data-scroll-target="#loading-data-in-r">Loading data in R</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data">Read in data</a></li>
  <li><a href="#gene-filtering" id="toc-gene-filtering" class="nav-link" data-scroll-target="#gene-filtering">Gene filtering</a></li>
  <li><a href="#qc-checks" id="toc-qc-checks" class="nav-link" data-scroll-target="#qc-checks">QC checks</a>
  <ul class="collapse">
  <li><a href="#expression-of-xist-gene" id="toc-expression-of-xist-gene" class="nav-link" data-scroll-target="#expression-of-xist-gene">Expression of <em>Xist</em> gene</a></li>
  <li><a href="#expression-of-y-chromosome-genes" id="toc-expression-of-y-chromosome-genes" class="nav-link" data-scroll-target="#expression-of-y-chromosome-genes">Expression of Y-chromosome genes</a></li>
  <li><a href="#principal-component-analysis" id="toc-principal-component-analysis" class="nav-link" data-scroll-target="#principal-component-analysis">Principal component analysis</a></li>
  </ul></li>
  <li><a href="#differential-expression-analysis" id="toc-differential-expression-analysis" class="nav-link" data-scroll-target="#differential-expression-analysis">Differential expression analysis</a></li>
  <li><a href="#gsea" id="toc-gsea" class="nav-link" data-scroll-target="#gsea">GSEA</a></li>
  <li><a href="#to-finish" id="toc-to-finish" class="nav-link" data-scroll-target="#to-finish">To finish</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="introduction" class="level1">
<h1>Introduction</h1>
</section>
<section id="loading-data-in-r" class="level1">
<h1>Loading data in R</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">## If packages are missing: install.packages('BiocManager')</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="do">## BiocManager::install('...')</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Setting up working directory</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="do">## color palettes</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>myPalette <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="do">## DE analysis</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Needed for GSEA</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(devtools<span class="sc">::</span><span class="fu">session_info</span>(), <span class="st">"session_info_limma.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-data" class="level1">
<h1>Read in data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load the prepared RangedSummarizedExperiment object and convert to DGEList</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>rse <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"rse_SRP218156.rds"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">SE2DGEList</span>(rse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="gene-filtering" class="level1">
<h1>Gene filtering</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Visualize distribution before filtering (protein-coding genes only)</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(edgeR<span class="sc">::</span><span class="fu">cpm</span>(dge, <span class="at">log =</span> T)[dge<span class="sc">$</span>genes<span class="sc">$</span>gene_type <span class="sc">%in%</span> <span class="st">"protein_coding"</span>,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    ], <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>tissue, <span class="at">col =</span> myPalette, <span class="at">legend =</span> <span class="st">"topright"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">## One cortex sample is weird: let's remove it</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[, <span class="fu">colnames</span>(dge) <span class="sc">!=</span> <span class="st">"ctx962"</span>]</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter lowly expressed genes</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(dge, <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>group)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>keep
FALSE  TRUE 
32710 22711 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## subset DGEList, while updating total counts</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> dge[keep, , keep.lib.sizes <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="do">## TMM normalization</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Calculate normalized log2(CPM) values</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>log2cpm <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(dge, <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">normalized.lib.sizes =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(log2cpm, <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>group, <span class="at">col =</span> myPalette, <span class="at">legend =</span> <span class="st">"topright"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDensities</span>(log2cpm, <span class="at">group =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>tissue, <span class="at">col =</span> myPalette, <span class="at">legend =</span> <span class="st">"topright"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-3-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="qc-checks" class="level1">
<h1>QC checks</h1>
<section id="expression-of-xist-gene" class="level2">
<h2 class="anchored" data-anchor-id="expression-of-xist-gene">Expression of <em>Xist</em> gene</h2>
<p>Xist is a lncRNA regulating the X-chromosome inactivation process in mammals, used to equalize the dosage of X-linked genes between female (XX) and male (XY). It should thus be expressed only in female samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm[<span class="fu">grep</span>(<span class="st">"Xist"</span>, dge<span class="sc">$</span>genes<span class="sc">$</span>gene_name), , drop <span class="ot">=</span> <span class="cn">FALSE</span>] <span class="sc">|&gt;</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">colour =</span> tissue, <span class="at">group =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="expression-of-y-chromosome-genes" class="level2">
<h2 class="anchored" data-anchor-id="expression-of-y-chromosome-genes">Expression of Y-chromosome genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Chromosome information</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(dge<span class="sc">$</span>genes<span class="sc">$</span>seqnames)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "chr1"       "chr2"       "chr3"       "chr4"       "chr5"      
 [6] "chr6"       "chr7"       "chr8"       "chr9"       "chr10"     
[11] "chr11"      "chr12"      "chr13"      "chr14"      "chr15"     
[16] "chr16"      "chr17"      "chr18"      "chr19"      "chrX"      
[21] "chrY"       "chrM"       "GL456210.1" "GL456211.1" "GL456212.1"
[26] "GL456216.1" "GL456219.1" "GL456221.1" "GL456233.1" "GL456239.1"
[31] "GL456350.1" "GL456354.1" "GL456372.1" "GL456381.1" "GL456385.1"
[36] "JH584292.1" "JH584293.1" "JH584294.1" "JH584295.1" "JH584296.1"
[41] "JH584297.1" "JH584298.1" "JH584299.1" "JH584303.1" "JH584304.1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## How many genes are left in our dataset after filtering?</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dge<span class="sc">$</span>genes<span class="sc">$</span>seqnames <span class="sc">==</span> <span class="st">"chrY"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
22700    11 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm[dge<span class="sc">$</span>genes<span class="sc">$</span>seqnames <span class="sc">==</span> <span class="st">"chrY"</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] <span class="sc">|&gt;</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">colour =</span> tissue, <span class="at">group =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">4</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> What do you observe? What is going on for some genes? (e.g., <em>Gm20775</em>)</p>
<p><img src="elemental-tip.png" class="img-fluid"> Bigwig files are available through recount3 for some of the samples and allow to visualize the read coverage (for example on the Y-chromosome) on a genome browser. For example you can provide these links to the UCSC genome browser directly:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(dge<span class="sc">$</span>samples<span class="sc">$</span>BigWigURL)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/14/sra.base_sums.SRP218156_SRR9956114.ALL.bw"
[2] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/15/sra.base_sums.SRP218156_SRR9956115.ALL.bw"
[3] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/16/sra.base_sums.SRP218156_SRR9956116.ALL.bw"
[4] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/17/sra.base_sums.SRP218156_SRR9956117.ALL.bw"
[5] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/18/sra.base_sums.SRP218156_SRR9956118.ALL.bw"
[6] "http://duffel.rail.bio/recount3/mouse/data_sources/sra/base_sums/56/SRP218156/19/sra.base_sums.SRP218156_SRR9956119.ALL.bw"</code></pre>
</div>
</div>
</section>
<section id="principal-component-analysis" class="level2">
<h2 class="anchored" data-anchor-id="principal-component-analysis">Principal component analysis</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## PCA on top 500 most variable genes</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>iqrs <span class="ot">&lt;-</span> <span class="fu">apply</span>(log2cpm, <span class="dv">1</span>, IQR)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>sel <span class="ot">&lt;-</span> iqrs <span class="sc">&gt;=</span> <span class="fu">sort</span>(iqrs, <span class="at">decreasing =</span> T)[<span class="dv">500</span>]</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>pca1 <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(log2cpm[sel, ]), <span class="at">scale =</span> T)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca1)<span class="sc">$</span>importance[, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                            PC1      PC2      PC3      PC4       PC5       PC6
Standard deviation     21.49999 3.859032 2.863773 1.589076 0.9230414 0.8300593
Proportion of Variance  0.92450 0.029780 0.016400 0.005050 0.0017000 0.0013800
Cumulative Proportion   0.92450 0.954280 0.970690 0.975740 0.9774400 0.9788200
                             PC7       PC8       PC9      PC10
Standard deviation     0.8055562 0.7751416 0.7446052 0.6975254
Proportion of Variance 0.0013000 0.0012000 0.0011100 0.0009700
Cumulative Proportion  0.9801200 0.9813200 0.9824300 0.9834000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Which sample metadata associate with which PCs?  Let's use the</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="do">## getExplanatoryPCs from scater package, which takes as input a</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="do">## SingleCellExperiment object containing dimensionality reduction results</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="at">logcounts =</span> log2cpm), <span class="at">reducedDims =</span> <span class="fu">list</span>(<span class="at">PCA =</span> pca1<span class="sc">$</span>x),</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">colData =</span> dge<span class="sc">$</span>samples)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">&lt;-</span> <span class="fu">getExplanatoryPCs</span>(sce, <span class="at">n_dimred =</span> <span class="dv">10</span>, <span class="at">dimred =</span> <span class="st">"PCA"</span>, <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"sex"</span>,</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"treatment"</span>, <span class="st">"tissue"</span>, <span class="st">"mouse"</span>))</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>percentVar</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              sex  treatment      tissue     mouse
PC1   0.006495126 0.01727742 99.93384555  0.503909
PC2   0.517791186 0.10868961 97.52726886  3.283735
PC3  97.542199169 1.22935290  0.25768486 97.896744
PC4   0.161111059 0.46281290  0.79569261 42.837992
PC5   0.676045162 0.13875380  0.04733970 26.723407
PC6   0.005913613 1.57780237  0.01936682 25.132588
PC7   0.251453219 0.27859827  0.02099720 34.903531
PC8   0.015834211 0.99585626  0.16509570 38.492577
PC9   0.001790200 0.54427412  0.06954958 35.126040
PC10  0.227016205 3.34983920  0.04426769 27.734207</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotExplanatoryPCs</span>(percentVar<span class="sc">/</span><span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-7-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> To which principal components contribute which factors? Does this corresponds to the expectations? What do you conclude on the relative effect size of different factors?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Visualize some of these associations, for example tissues with PCs 1 and 2,</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="do">## sex with PC3</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca1<span class="sc">$</span>x[, <span class="dv">1</span>], pca1<span class="sc">$</span>x[, <span class="dv">2</span>], <span class="at">pch =</span> <span class="fu">c</span>(<span class="dv">15</span><span class="sc">:</span><span class="dv">16</span>)[dge<span class="sc">$</span>samples<span class="sc">$</span>treatment], <span class="at">col =</span> myPalette[dge<span class="sc">$</span>samples<span class="sc">$</span>tissue])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca1<span class="sc">$</span>x[, <span class="dv">2</span>], pca1<span class="sc">$</span>x[, <span class="dv">3</span>], <span class="at">pch =</span> <span class="fu">c</span>(<span class="dv">15</span><span class="sc">:</span><span class="dv">16</span>)[dge<span class="sc">$</span>samples<span class="sc">$</span>treatment], <span class="at">col =</span> myPalette[dge<span class="sc">$</span>samples<span class="sc">$</span>sex])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis" class="level1">
<h1>Differential expression analysis</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Design matrix We use an additive model accounting for tissue effect (which</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="do">## is not of main interest here)</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>moma <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> group <span class="sc">+</span> tissue, <span class="at">data =</span> dge<span class="sc">$</span>samples)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(moma) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"group"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(moma))  <span class="do">## Easier to manipulate contrasts</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Contrast matrix: - Direct male vs. female comparison (in controls only) -</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="do">## effect of Tamoxifen treatment in each sex separately - interaction term to</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="do">## extract genes reacting differently to Tamoxifen treatment in male and female</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>contrasts.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">Male_vs_female =</span> Vehicle.male <span class="sc">-</span> Vehicle.female,</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">Tamoxifen_effet_male =</span> Tamoxifen.male <span class="sc">-</span> Vehicle.male, <span class="at">Tamoxifen_effet_female =</span> Tamoxifen.female <span class="sc">-</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>        Vehicle.female, <span class="at">Interaction =</span> (Tamoxifen.male <span class="sc">-</span> Vehicle.male) <span class="sc">-</span> (Tamoxifen.female <span class="sc">-</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>        Vehicle.female), <span class="at">levels =</span> moma)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Run DE analysis</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, moma, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, moma)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, contrasts.matrix)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Extract genes significant at FDR 10%</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(fit2, <span class="at">p.value =</span> <span class="fl">0.1</span>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(de, <span class="dv">2</span>, table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$Male_vs_female

   -1     0     1 
   36 22663    12 

$Tamoxifen_effet_male

   -1     0     1 
   11 22685    15 

$Tamoxifen_effet_female

   -1     0     1 
  113 22564    34 

$Interaction

    0     1 
22709     2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Top genes for each contrast</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Male_vs_female"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand  source type
ENSMUSG00000056673     chrY    897788    956786  58999      +  HAVANA gene
ENSMUSG00000035150     chrX  94188707  94212862  24156      -  HAVANA gene
ENSMUSG00000069045     chrY   1260771   1286629  25859      -  HAVANA gene
ENSMUSG00000068457     chrY   1096861   1245759 148899      -  HAVANA gene
ENSMUSG00000099312     chrX 103461218 103461285     68      - ENSEMBL gene
ENSMUSG00000086503     chrX 103460366 103483254  22889      -  HAVANA gene
ENSMUSG00000098262     chrX 103480946 103481026     81      + ENSEMBL gene
ENSMUSG00000069049     chrY   1010543   1028847  18305      +  HAVANA gene
ENSMUSG00000098743     chrX 103469671 103469796    126      - ENSEMBL gene
ENSMUSG00000099876     chrY   1048393   1049134    742      +  HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000056673     15164    NA ENSMUSG00000056673 protein_coding     Kdm5d
ENSMUSG00000035150      3812    NA ENSMUSG00000035150 protein_coding   Eif2s3x
ENSMUSG00000069045      5199    NA ENSMUSG00000069045 protein_coding     Ddx3y
ENSMUSG00000068457      9791    NA ENSMUSG00000068457 protein_coding       Uty
ENSMUSG00000099312        68    NA ENSMUSG00000099312       misc_RNA   Gm27733
ENSMUSG00000086503     17946    NA ENSMUSG00000086503         lncRNA      Xist
ENSMUSG00000098262        81    NA ENSMUSG00000098262       misc_RNA   Gm27520
ENSMUSG00000069049      4161    NA ENSMUSG00000069049 protein_coding   Eif2s3y
ENSMUSG00000098743       126    NA ENSMUSG00000098743       misc_RNA   Gm27927
ENSMUSG00000099876       742    NA ENSMUSG00000099876         lncRNA   Gm29650
                   level      mgi_id          havana_gene               tag
ENSMUSG00000056673     2   MGI:99780 OTTMUSG00000045277.1              &lt;NA&gt;
ENSMUSG00000035150     2 MGI:1349431 OTTMUSG00000018084.1              &lt;NA&gt;
ENSMUSG00000069045     2 MGI:1349406 OTTMUSG00000045279.1 overlapping_locus
ENSMUSG00000068457     2  MGI:894810 OTTMUSG00000032814.5              &lt;NA&gt;
ENSMUSG00000099312     3 MGI:5531115                 &lt;NA&gt;              &lt;NA&gt;
ENSMUSG00000086503     2   MGI:98974 OTTMUSG00000018295.5        ncRNA_host
ENSMUSG00000098262     3 MGI:5530902                 &lt;NA&gt;              &lt;NA&gt;
ENSMUSG00000069049     2 MGI:1349430 OTTMUSG00000021559.1              &lt;NA&gt;
ENSMUSG00000098743     3 MGI:5531309                 &lt;NA&gt;              &lt;NA&gt;
ENSMUSG00000099876     2 MGI:5580356 OTTMUSG00000045278.1              &lt;NA&gt;
                         logFC    AveExpr         t      P.Value    adj.P.Val
ENSMUSG00000056673   9.5462260 -0.4319924  16.37566 7.502118e-21 9.202729e-17
ENSMUSG00000035150  -0.6523096  6.2281911 -16.34369 8.104204e-21 9.202729e-17
ENSMUSG00000069045  11.0668444  0.2849525  15.72543 3.684101e-20 2.360479e-16
ENSMUSG00000068457  10.3835065 -0.7664206  15.64466 4.503799e-20 2.360479e-16
ENSMUSG00000099312  -7.7367972 -2.8949791 -15.58728 5.196775e-20 2.360479e-16
ENSMUSG00000086503 -10.2229207  3.6571905 -15.34143 9.635731e-20 3.647285e-16
ENSMUSG00000098262  -7.0767101 -3.2312004 -14.69633 5.029983e-19 1.631942e-15
ENSMUSG00000069049  11.0428693 -0.4760528  12.96064 5.478409e-17 1.555252e-13
ENSMUSG00000098743  -8.7126677 -2.2759046 -12.53707 1.820954e-16 4.595077e-13
ENSMUSG00000099876   6.5517579 -3.1854202  11.81076 1.506397e-15 3.421178e-12
                          B
ENSMUSG00000056673 19.54949
ENSMUSG00000035150 37.19013
ENSMUSG00000069045 18.37094
ENSMUSG00000068457 17.51796
ENSMUSG00000099312 20.42205
ENSMUSG00000086503 32.67395
ENSMUSG00000098262 19.28454
ENSMUSG00000069049 15.01886
ENSMUSG00000098743 16.97675
ENSMUSG00000099876 13.67767</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Tamoxifen_effet_male"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000022602    chr15  74669083  74672570   3488      - HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000025324     chr7  58656166  58829420 173255      + HAVANA gene
ENSMUSG00000025316     chr8 121949750 122029258  79509      + HAVANA gene
ENSMUSG00000032193     chr9  21723483  21749919  26437      + HAVANA gene
ENSMUSG00000052837     chr8  84974484  84978718   4235      - HAVANA gene
ENSMUSG00000008090     chr5 108692382 108706924  14543      + HAVANA gene
ENSMUSG00000023034    chr15 101254269 101274795  20527      + HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000022602      3057    NA ENSMUSG00000022602 protein_coding       Arc
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000025324      5587    NA ENSMUSG00000025324 protein_coding    Atp10a
ENSMUSG00000025316      6632    NA ENSMUSG00000025316 protein_coding      Banp
ENSMUSG00000032193      9622    NA ENSMUSG00000032193 protein_coding      Ldlr
ENSMUSG00000052837      2301    NA ENSMUSG00000052837 protein_coding      Junb
ENSMUSG00000008090      2782    NA ENSMUSG00000008090 protein_coding    Fgfrl1
ENSMUSG00000023034      3509    NA ENSMUSG00000023034 protein_coding     Nr4a1
                   level      mgi_id          havana_gene               tag
ENSMUSG00000022602     2   MGI:88067 OTTMUSG00000016386.3              &lt;NA&gt;
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3              &lt;NA&gt;
ENSMUSG00000025324     2 MGI:1330809 OTTMUSG00000058914.1              &lt;NA&gt;
ENSMUSG00000025316     1 MGI:1889023 OTTMUSG00000036899.8 overlapping_locus
ENSMUSG00000032193     2   MGI:96765 OTTMUSG00000063513.2              &lt;NA&gt;
ENSMUSG00000052837     2   MGI:96647 OTTMUSG00000061397.1              &lt;NA&gt;
ENSMUSG00000008090     2 MGI:2150920 OTTMUSG00000054022.2              &lt;NA&gt;
ENSMUSG00000023034     2 MGI:1352454 OTTMUSG00000071152.1              &lt;NA&gt;
                        logFC  AveExpr         t      P.Value    adj.P.Val
ENSMUSG00000022602  0.5785466 5.636934 11.007342 1.688210e-14 3.834094e-10
ENSMUSG00000066687 -0.8031728 4.422495 -9.771783 8.154891e-13 9.260286e-09
ENSMUSG00000021250  0.5426891 4.877409  7.383113 2.389416e-09 1.808867e-05
ENSMUSG00000037868  0.7340948 0.373090  6.602602 3.551604e-08 2.016512e-04
ENSMUSG00000025324 -0.5365920 2.959977 -6.470752 5.610024e-08 2.548185e-04
ENSMUSG00000025316 -0.3640463 4.583836 -6.002141 2.843207e-07 1.076201e-03
ENSMUSG00000032193  0.4048792 4.676372  5.587792 1.184838e-06 3.844121e-03
ENSMUSG00000052837  0.5331572 4.998842  5.392233 2.313098e-06 6.566595e-03
ENSMUSG00000008090  0.3314394 3.320005  5.325943 2.899289e-06 6.912983e-03
ENSMUSG00000023034  0.3736489 5.804234  5.311641 3.043892e-06 6.912983e-03
                           B
ENSMUSG00000022602 22.791293
ENSMUSG00000066687 18.854088
ENSMUSG00000021250 11.278945
ENSMUSG00000037868  8.378304
ENSMUSG00000025324  8.084094
ENSMUSG00000025316  6.709000
ENSMUSG00000032193  5.347324
ENSMUSG00000052837  4.676239
ENSMUSG00000008090  4.515779
ENSMUSG00000023034  4.358321</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Tamoxifen_effet_female"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000025316     chr8 121949750 122029258  79509      + HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000004328     chr7  17031507  17062427  30921      - HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000020484    chr11   5520659   5525893   5235      + HAVANA gene
ENSMUSG00000022602    chr15  74669083  74672570   3488      - HAVANA gene
ENSMUSG00000023034    chr15 101254269 101274795  20527      + HAVANA gene
ENSMUSG00000054720     chr5 105519388 105613018  93631      + HAVANA gene
ENSMUSG00000035877     chr2 160748708 160872998 124291      - HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000025316      6632    NA ENSMUSG00000025316 protein_coding      Banp
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000004328      6710    NA ENSMUSG00000004328 protein_coding     Hif3a
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000020484      3248    NA ENSMUSG00000020484 protein_coding      Xbp1
ENSMUSG00000022602      3057    NA ENSMUSG00000022602 protein_coding       Arc
ENSMUSG00000023034      3509    NA ENSMUSG00000023034 protein_coding     Nr4a1
ENSMUSG00000054720      7654    NA ENSMUSG00000054720 protein_coding    Lrrc8c
ENSMUSG00000035877     13259    NA ENSMUSG00000035877 protein_coding      Zhx3
                   level      mgi_id          havana_gene               tag
ENSMUSG00000025316     1 MGI:1889023 OTTMUSG00000036899.8 overlapping_locus
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3              &lt;NA&gt;
ENSMUSG00000004328     2 MGI:1859778 OTTMUSG00000022817.2              &lt;NA&gt;
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
ENSMUSG00000020484     2   MGI:98970 OTTMUSG00000006338.4              &lt;NA&gt;
ENSMUSG00000022602     2   MGI:88067 OTTMUSG00000016386.3              &lt;NA&gt;
ENSMUSG00000023034     2 MGI:1352454 OTTMUSG00000071152.1              &lt;NA&gt;
ENSMUSG00000054720     2 MGI:2140839 OTTMUSG00000026735.2 overlapping_locus
ENSMUSG00000035877     2 MGI:2444772 OTTMUSG00000001048.8 overlapping_locus
                        logFC  AveExpr          t      P.Value    adj.P.Val
ENSMUSG00000025316 -0.6833900 4.583836 -10.571999 6.474533e-14 1.470431e-09
ENSMUSG00000037868  1.0153592 0.373090   9.252371 4.402047e-12 4.998744e-08
ENSMUSG00000004328 -1.1705553 2.342140  -8.388406 7.770217e-11 5.882313e-07
ENSMUSG00000066687 -0.7037366 4.422495  -8.071347 2.269733e-10 1.229943e-06
ENSMUSG00000021250  0.6150187 4.877409   8.019393 2.707812e-10 1.229943e-06
ENSMUSG00000020484 -0.3250517 6.047353  -7.191641 4.623392e-09 1.750031e-05
ENSMUSG00000022602  0.3806623 5.636934   6.854151 1.485341e-08 4.336124e-05
ENSMUSG00000023034  0.5092910 5.804234   6.846087 1.527409e-08 4.336124e-05
ENSMUSG00000054720 -0.4309182 4.741372  -6.621467 3.326749e-08 7.973826e-05
ENSMUSG00000035877 -0.2767113 6.115502  -6.605919 3.510997e-08 7.973826e-05
                           B
ENSMUSG00000025316 21.223606
ENSMUSG00000037868 16.692353
ENSMUSG00000004328 13.343961
ENSMUSG00000066687 13.545944
ENSMUSG00000021250 13.388002
ENSMUSG00000020484 10.623886
ENSMUSG00000022602  9.451315
ENSMUSG00000023034  9.459263
ENSMUSG00000054720  8.748828
ENSMUSG00000035877  8.655949</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000004328     chr7  17031507  17062427  30921      - HAVANA gene
ENSMUSG00000037754     chr2 158665398 158766334 100937      + HAVANA gene
ENSMUSG00000052040     chr7  63886351  63938915  52565      - HAVANA gene
ENSMUSG00000105843     chr3  88030878  88033150   2273      + HAVANA gene
ENSMUSG00000031431     chrX 140539528 140600659  61132      - HAVANA gene
ENSMUSG00000030168     chr6 119353150 119417704  64555      - HAVANA gene
ENSMUSG00000028655     chr4 122946850 122961188  14339      - HAVANA gene
ENSMUSG00000039911     chr4 149896283 149955043  58761      - HAVANA gene
ENSMUSG00000035104     chr6  82041043  82093099  52057      + HAVANA gene
ENSMUSG00000051335    chr13  43195245  43304172 108928      - HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000004328      6710    NA ENSMUSG00000004328 protein_coding     Hif3a
ENSMUSG00000037754      8844    NA ENSMUSG00000037754 protein_coding  Ppp1r16b
ENSMUSG00000052040      7531    NA ENSMUSG00000052040 protein_coding     Klf13
ENSMUSG00000105843      2273    NA ENSMUSG00000105843            TEC   Gm19439
ENSMUSG00000031431      4341    NA ENSMUSG00000031431 protein_coding   Tsc22d3
ENSMUSG00000030168      5979    NA ENSMUSG00000030168 protein_coding   Adipor2
ENSMUSG00000028655      2250    NA ENSMUSG00000028655 protein_coding    Mfsd2a
ENSMUSG00000039911      4396    NA ENSMUSG00000039911 protein_coding     Spsb1
ENSMUSG00000035104      2805    NA ENSMUSG00000035104 protein_coding     Eva1a
ENSMUSG00000051335      7208    NA ENSMUSG00000051335 protein_coding     Gfod1
                   level      mgi_id          havana_gene               tag
ENSMUSG00000004328     2 MGI:1859778 OTTMUSG00000022817.2              &lt;NA&gt;
ENSMUSG00000037754     2 MGI:2151841 OTTMUSG00000015931.2        ncRNA_host
ENSMUSG00000052040     2 MGI:1354948 OTTMUSG00000044885.7              &lt;NA&gt;
ENSMUSG00000105843     2 MGI:5011624 OTTMUSG00000052443.1              &lt;NA&gt;
ENSMUSG00000031431     2 MGI:1196284 OTTMUSG00000018845.2              &lt;NA&gt;
ENSMUSG00000030168     2   MGI:93830 OTTMUSG00000049368.6 overlapping_locus
ENSMUSG00000028655     2 MGI:1923824 OTTMUSG00000008939.2              &lt;NA&gt;
ENSMUSG00000039911     2 MGI:1921896 OTTMUSG00000010263.5              &lt;NA&gt;
ENSMUSG00000035104     2 MGI:2385247 OTTMUSG00000027004.3              &lt;NA&gt;
ENSMUSG00000051335     2 MGI:2145304 OTTMUSG00000067253.1              &lt;NA&gt;
                       logFC     AveExpr        t      P.Value  adj.P.Val
ENSMUSG00000004328 1.0335209  2.34214032 5.422892 2.083304e-06 0.04731391
ENSMUSG00000037754 0.2994358  6.16239621 5.033503 7.802387e-06 0.08860001
ENSMUSG00000052040 0.2902667  6.65749111 4.357083 7.288062e-05 0.50247264
ENSMUSG00000105843 0.6438192 -0.06401852 4.296843 8.849855e-05 0.50247264
ENSMUSG00000031431 0.3650196  5.93605236 4.088143 1.721256e-04 0.66374346
ENSMUSG00000030168 0.2198804  6.25129810 4.080618 1.762640e-04 0.66374346
ENSMUSG00000028655 0.4493447  3.25778952 4.024161 2.105447e-04 0.66374346
ENSMUSG00000039911 0.4317396  3.37968541 3.990728 2.338051e-04 0.66374346
ENSMUSG00000035104 0.4523403  2.53485835 3.890004 3.199267e-04 0.77062375
ENSMUSG00000051335 0.2337552  6.48530078 3.867813 3.426630e-04 0.77062375
                            B
ENSMUSG00000004328 -0.8990718
ENSMUSG00000037754  2.5073164
ENSMUSG00000052040  0.9671582
ENSMUSG00000105843 -1.7914496
ENSMUSG00000031431  0.2009658
ENSMUSG00000030168  0.1922932
ENSMUSG00000028655 -0.8772917
ENSMUSG00000039911 -0.7684920
ENSMUSG00000035104 -1.9008504
ENSMUSG00000051335 -0.1932112</code></pre>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> Do the number of DE genes correspond to your expectations from the PCA? What would you conclude on the difference of Tamoxifen effects in male and female?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot expression of top genes</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Since we account for tissue, maybe it makes sense to correct the logCPM for</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="do">## this effect</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>log2cpm.corrected <span class="ot">&lt;-</span> <span class="fu">removeBatchEffect</span>(log2cpm, <span class="at">batch =</span> dge<span class="sc">$</span>samples<span class="sc">$</span>tissue)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="do">## TO DO: add design?  design matrix of interest not specified. Assuming a</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="do">## one-group experiment.</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Sex differences, top 10 genes</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Male_vs_female"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm.corrected[top<span class="sc">$</span>gene_id, ] <span class="sc">|&gt;</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm.corrected), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">group =</span> tissue, <span class="at">col =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> Which genes do you notice here? Try and plot the original logCPM values (uncorrected for tissue effects)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Male_vs_female"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">factor</span>(top<span class="sc">$</span>seqname))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
chrX chrY 
   5    5 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Treatment in males, top 10 genes</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Tamoxifen_effet_male"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm.corrected[top<span class="sc">$</span>gene_id, ] <span class="sc">|&gt;</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm.corrected), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">group =</span> tissue, <span class="at">col =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Treatment in females, top 10 genes</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Tamoxifen_effet_female"</span>, <span class="at">n =</span> <span class="dv">10</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm.corrected[top<span class="sc">$</span>gene_id, ] <span class="sc">|&gt;</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm.corrected), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">group =</span> tissue, <span class="at">col =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> What do you notice? Are some of these genes described in the original paper?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="do">## See Fig 4e and f</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Tamoxifen_effet_male"</span>, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">n =</span> <span class="cn">Inf</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>top[top<span class="sc">$</span>gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Egr2"</span>, <span class="st">"Fos"</span>, <span class="st">"Dusp1"</span>, <span class="st">"Nr4a1"</span>, <span class="st">"Sik1"</span>, <span class="st">"Arc"</span>, <span class="st">"Egr1"</span>,</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Plcl2"</span>, <span class="st">"Galnt9"</span>, <span class="st">"Per2"</span>, <span class="st">"Zbtb16"</span>, <span class="st">"Map3k13"</span>, <span class="st">"Banp"</span>), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000022602    chr15  74669083  74672570   3488      - HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000025316     chr8 121949750 122029258  79509      + HAVANA gene
ENSMUSG00000023034    chr15 101254269 101274795  20527      + HAVANA gene
ENSMUSG00000038418    chr18  34859823  34864984   5162      + HAVANA gene
ENSMUSG00000024042    chr17  31844250  31855804  11555      - HAVANA gene
ENSMUSG00000024190    chr17  26505590  26562128  56539      - HAVANA gene
ENSMUSG00000055866     chr1  91415982  91459324  43343      - HAVANA gene
ENSMUSG00000038910    chr17  50509403  50688484 179082      + HAVANA gene
ENSMUSG00000033618    chr16  21794346  21933439 139094      + HAVANA gene
ENSMUSG00000033316     chr5 110544355 110621380  77026      + HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000022602      3057    NA ENSMUSG00000022602 protein_coding       Arc
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000025316      6632    NA ENSMUSG00000025316 protein_coding      Banp
ENSMUSG00000023034      3509    NA ENSMUSG00000023034 protein_coding     Nr4a1
ENSMUSG00000038418      4484    NA ENSMUSG00000038418 protein_coding      Egr1
ENSMUSG00000024042      5133    NA ENSMUSG00000024042 protein_coding      Sik1
ENSMUSG00000024190      3311    NA ENSMUSG00000024190 protein_coding     Dusp1
ENSMUSG00000055866      5856    NA ENSMUSG00000055866 protein_coding      Per2
ENSMUSG00000038910      6626    NA ENSMUSG00000038910 protein_coding     Plcl2
ENSMUSG00000033618      8714    NA ENSMUSG00000033618 protein_coding   Map3k13
ENSMUSG00000033316      3006    NA ENSMUSG00000033316 protein_coding    Galnt9
                   level      mgi_id          havana_gene               tag
ENSMUSG00000022602     2   MGI:88067 OTTMUSG00000016386.3              &lt;NA&gt;
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3              &lt;NA&gt;
ENSMUSG00000025316     1 MGI:1889023 OTTMUSG00000036899.8 overlapping_locus
ENSMUSG00000023034     2 MGI:1352454 OTTMUSG00000071152.1              &lt;NA&gt;
ENSMUSG00000038418     2   MGI:95295 OTTMUSG00000021064.2              &lt;NA&gt;
ENSMUSG00000024042     2  MGI:104754 OTTMUSG00000073186.1              &lt;NA&gt;
ENSMUSG00000024190     2  MGI:105120 OTTMUSG00000020083.4              &lt;NA&gt;
ENSMUSG00000055866     2 MGI:1195265 OTTMUSG00000048493.1              &lt;NA&gt;
ENSMUSG00000038910     2 MGI:1352756 OTTMUSG00000072532.1              &lt;NA&gt;
ENSMUSG00000033618     2 MGI:2444243 OTTMUSG00000071512.1 overlapping_locus
ENSMUSG00000033316     2 MGI:2677965 OTTMUSG00000054216.1              &lt;NA&gt;
                         logFC  AveExpr          t      P.Value    adj.P.Val
ENSMUSG00000022602  0.57854660 5.636934 11.0073419 1.688210e-14 3.834094e-10
ENSMUSG00000066687 -0.80317278 4.422495 -9.7717826 8.154891e-13 9.260286e-09
ENSMUSG00000021250  0.54268914 4.877409  7.3831131 2.389416e-09 1.808867e-05
ENSMUSG00000037868  0.73409479 0.373090  6.6026018 3.551604e-08 2.016512e-04
ENSMUSG00000025316 -0.36404627 4.583836 -6.0021409 2.843207e-07 1.076201e-03
ENSMUSG00000023034  0.37364885 5.804234  5.3116406 3.043892e-06 6.912983e-03
ENSMUSG00000038418  0.40372773 6.923094  5.2003157 4.442075e-06 9.171269e-03
ENSMUSG00000024042  0.38386485 3.747410  4.0988120 1.664194e-04 1.219210e-01
ENSMUSG00000024190  0.28850524 4.473001  3.9713129 2.484360e-04 1.513791e-01
ENSMUSG00000055866 -0.24828732 4.916803 -3.3971385 1.410866e-03 2.921330e-01
ENSMUSG00000038910 -0.14392733 5.382672 -2.7821041 7.797423e-03 4.786143e-01
ENSMUSG00000033618 -0.11837673 5.487564 -2.2477017 2.941671e-02 6.129810e-01
ENSMUSG00000033316 -0.06488197 5.498505 -0.9928178 3.259762e-01 8.945678e-01
                            B
ENSMUSG00000022602 22.7912932
ENSMUSG00000066687 18.8540878
ENSMUSG00000021250 11.2789455
ENSMUSG00000037868  8.3783045
ENSMUSG00000025316  6.7090000
ENSMUSG00000023034  4.3583209
ENSMUSG00000038418  3.9089363
ENSMUSG00000024042  0.7545411
ENSMUSG00000024190  0.2804978
ENSMUSG00000055866 -1.3515187
ENSMUSG00000038910 -3.0188769
ENSMUSG00000033618 -4.1313932
ENSMUSG00000033316 -6.1456138</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Tamoxifen_effet_female"</span>, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">n =</span> <span class="cn">Inf</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>top[top<span class="sc">$</span>gene_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Egr2"</span>, <span class="st">"Fos"</span>, <span class="st">"Dusp1"</span>, <span class="st">"Nr4a1"</span>, <span class="st">"Sik1"</span>, <span class="st">"Arc"</span>, <span class="st">"Egr1"</span>,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Plcl2"</span>, <span class="st">"Galnt9"</span>, <span class="st">"Per2"</span>, <span class="st">"Zbtb16"</span>, <span class="st">"Map3k13"</span>, <span class="st">"Banp"</span>), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   seqnames     start       end  width strand source type
ENSMUSG00000025316     chr8 121949750 122029258  79509      + HAVANA gene
ENSMUSG00000037868    chr10  67535475  67542188   6714      + HAVANA gene
ENSMUSG00000066687     chr9  48654297  48836222 181926      - HAVANA gene
ENSMUSG00000021250    chr12  85473890  85477273   3384      + HAVANA gene
ENSMUSG00000022602    chr15  74669083  74672570   3488      - HAVANA gene
ENSMUSG00000023034    chr15 101254269 101274795  20527      + HAVANA gene
ENSMUSG00000055866     chr1  91415982  91459324  43343      - HAVANA gene
ENSMUSG00000033618    chr16  21794346  21933439 139094      + HAVANA gene
ENSMUSG00000024190    chr17  26505590  26562128  56539      - HAVANA gene
ENSMUSG00000038418    chr18  34859823  34864984   5162      + HAVANA gene
ENSMUSG00000033316     chr5 110544355 110621380  77026      + HAVANA gene
ENSMUSG00000038910    chr17  50509403  50688484 179082      + HAVANA gene
ENSMUSG00000024042    chr17  31844250  31855804  11555      - HAVANA gene
                   bp_length phase            gene_id      gene_type gene_name
ENSMUSG00000025316      6632    NA ENSMUSG00000025316 protein_coding      Banp
ENSMUSG00000037868      3398    NA ENSMUSG00000037868 protein_coding      Egr2
ENSMUSG00000066687      5283    NA ENSMUSG00000066687 protein_coding    Zbtb16
ENSMUSG00000021250      2352    NA ENSMUSG00000021250 protein_coding       Fos
ENSMUSG00000022602      3057    NA ENSMUSG00000022602 protein_coding       Arc
ENSMUSG00000023034      3509    NA ENSMUSG00000023034 protein_coding     Nr4a1
ENSMUSG00000055866      5856    NA ENSMUSG00000055866 protein_coding      Per2
ENSMUSG00000033618      8714    NA ENSMUSG00000033618 protein_coding   Map3k13
ENSMUSG00000024190      3311    NA ENSMUSG00000024190 protein_coding     Dusp1
ENSMUSG00000038418      4484    NA ENSMUSG00000038418 protein_coding      Egr1
ENSMUSG00000033316      3006    NA ENSMUSG00000033316 protein_coding    Galnt9
ENSMUSG00000038910      6626    NA ENSMUSG00000038910 protein_coding     Plcl2
ENSMUSG00000024042      5133    NA ENSMUSG00000024042 protein_coding      Sik1
                   level      mgi_id          havana_gene               tag
ENSMUSG00000025316     1 MGI:1889023 OTTMUSG00000036899.8 overlapping_locus
ENSMUSG00000037868     2   MGI:95296 OTTMUSG00000033571.3              &lt;NA&gt;
ENSMUSG00000066687     2  MGI:103222 OTTMUSG00000063009.1              &lt;NA&gt;
ENSMUSG00000021250     2   MGI:95574 OTTMUSG00000019764.1              &lt;NA&gt;
ENSMUSG00000022602     2   MGI:88067 OTTMUSG00000016386.3              &lt;NA&gt;
ENSMUSG00000023034     2 MGI:1352454 OTTMUSG00000071152.1              &lt;NA&gt;
ENSMUSG00000055866     2 MGI:1195265 OTTMUSG00000048493.1              &lt;NA&gt;
ENSMUSG00000033618     2 MGI:2444243 OTTMUSG00000071512.1 overlapping_locus
ENSMUSG00000024190     2  MGI:105120 OTTMUSG00000020083.4              &lt;NA&gt;
ENSMUSG00000038418     2   MGI:95295 OTTMUSG00000021064.2              &lt;NA&gt;
ENSMUSG00000033316     2 MGI:2677965 OTTMUSG00000054216.1              &lt;NA&gt;
ENSMUSG00000038910     2 MGI:1352756 OTTMUSG00000072532.1              &lt;NA&gt;
ENSMUSG00000024042     2  MGI:104754 OTTMUSG00000073186.1              &lt;NA&gt;
                        logFC  AveExpr          t      P.Value    adj.P.Val
ENSMUSG00000025316 -0.6833900 4.583836 -10.571999 6.474533e-14 1.470431e-09
ENSMUSG00000037868  1.0153592 0.373090   9.252371 4.402047e-12 4.998744e-08
ENSMUSG00000066687 -0.7037366 4.422495  -8.071347 2.269733e-10 1.229943e-06
ENSMUSG00000021250  0.6150187 4.877409   8.019393 2.707812e-10 1.229943e-06
ENSMUSG00000022602  0.3806623 5.636934   6.854151 1.485341e-08 4.336124e-05
ENSMUSG00000023034  0.5092910 5.804234   6.846087 1.527409e-08 4.336124e-05
ENSMUSG00000055866 -0.4589804 4.916803  -5.907935 3.936682e-07 4.967000e-04
ENSMUSG00000033618 -0.2974800 5.487564  -5.344778 2.719204e-06 2.287253e-03
ENSMUSG00000024190  0.3652424 4.473001   4.820397 1.592442e-05 7.753215e-03
ENSMUSG00000038418  0.3794482 6.923094   4.637493 2.918620e-05 1.250656e-02
ENSMUSG00000033316 -0.2092952 5.498505  -3.024000 4.065511e-03 2.642152e-01
ENSMUSG00000038910 -0.1519860 5.382672  -2.781750 7.804682e-03 3.489753e-01
ENSMUSG00000024042  0.2535749 3.747410   2.590761 1.277603e-02 4.233239e-01
                           B
ENSMUSG00000025316 21.223606
ENSMUSG00000037868 16.692353
ENSMUSG00000066687 13.545944
ENSMUSG00000021250 13.388002
ENSMUSG00000022602  9.451315
ENSMUSG00000023034  9.459263
ENSMUSG00000055866  6.372583
ENSMUSG00000033618  4.492261
ENSMUSG00000024190  2.827890
ENSMUSG00000038418  2.026313
ENSMUSG00000033316 -2.519095
ENSMUSG00000038910 -3.089932
ENSMUSG00000024042 -3.248676</code></pre>
</div>
</div>
<p>We will now look at the 2 genes reacting differently to Tamoxifen treatment in male and female</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction"</span>, <span class="at">p.value =</span> <span class="fl">0.1</span>, <span class="at">sort.by =</span> <span class="st">"P"</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> log2cpm.corrected[top<span class="sc">$</span>gene_id, ] <span class="sc">|&gt;</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="cn">NA</span>) <span class="sc">|&gt;</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> rowname) <span class="sc">|&gt;</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">colnames</span>(log2cpm.corrected), <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"log2CPM"</span>) <span class="sc">|&gt;</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>samples), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Sample"</span> <span class="sc">==</span> <span class="st">"sra.sample_title"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="at">y =</span> <span class="fu">as_tibble</span>(dge<span class="sc">$</span>genes), <span class="at">by =</span> <span class="fu">join_by</span>(<span class="st">"Gene"</span> <span class="sc">==</span> <span class="st">"gene_id"</span>))</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> log2CPM, <span class="at">group =</span> tissue, <span class="at">col =</span> tissue)) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene_name,</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> myPalette[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="hands_on_limma_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><img src="round-help-button.png" class="img-fluid"> How would you describe the behavior of these genes in (well chosen) simple words? Are these genes described in the original paper?</p>
</section>
<section id="gsea" class="level1">
<h1>GSEA</h1>
<p>It is possible that we detect few significant genes for the interaction but that some trends are visible collectively at the gene set level</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>GOmapping <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Mm.eg.db, <span class="at">keys =</span> dge<span class="sc">$</span>genes<span class="sc">$</span>gene_id, <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GOALL"</span>, <span class="st">"EVIDENCEALL"</span>, <span class="st">"ONTOLOGYALL"</span>))</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Transform to list</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>GOmapping <span class="ot">&lt;-</span> <span class="fu">tapply</span>(GOmapping<span class="sc">$</span>ENSEMBL, <span class="fu">list</span>(GOmapping<span class="sc">$</span>GOALL), unique)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="do">## process and filter gene sets</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>indx <span class="ot">&lt;-</span> <span class="fu">ids2indices</span>(<span class="at">gene.sets =</span> GOmapping, <span class="at">identifiers =</span> dge<span class="sc">$</span>genes<span class="sc">$</span>gene_id)</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Filter gene sets smaller than 10 genes, and gene sets larger than 500 genes</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>indx <span class="ot">&lt;-</span> indx[<span class="fu">sapply</span>(indx, length) <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="sc">&amp;</span> <span class="fu">sapply</span>(indx, length) <span class="sc">&lt;</span> <span class="dv">500</span>]</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Run CameraPR on the t-statistics</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">coef =</span> <span class="st">"Interaction"</span>, <span class="at">p.value =</span> <span class="dv">1</span>, <span class="at">n =</span> <span class="cn">Inf</span>, <span class="at">sort.by =</span> <span class="st">"none"</span>)</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cameraPR</span>(<span class="at">statistic =</span> top<span class="sc">$</span>t, <span class="at">index =</span> indx)</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Add info on Gene Ontology terms</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>res_gsea <span class="ot">&lt;-</span> <span class="fu">cbind</span>(res_gsea, AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(GO.db, <span class="at">keys =</span> <span class="fu">row.names</span>(res_gsea),</span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype =</span> <span class="st">"GOID"</span>, <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"ONTOLOGY"</span>, <span class="st">"TERM"</span>)))  <span class="co">#, 'DEFINITION')))</span></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(res_gsea<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 8166    44 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_gsea, <span class="at">n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           NGenes Direction       PValue        FDR       GOID ONTOLOGY
GO:0099010     47        Up 5.789342e-06 0.01784664 GO:0099010       BP
GO:0060412     46        Up 9.062625e-06 0.01784664 GO:0060412       BP
GO:0046703     16      Down 1.275257e-05 0.01784664 GO:0046703       MF
GO:0042832     32      Down 1.417637e-05 0.01784664 GO:0042832       BP
GO:0099563     55        Up 1.431596e-05 0.01784664 GO:0099563       BP
GO:0002483     21      Down 1.777337e-05 0.01784664 GO:0002483       BP
GO:0019885     21      Down 1.777337e-05 0.01784664 GO:0019885       BP
GO:0002476     16      Down 2.164181e-05 0.01784664 GO:0002476       BP
GO:0002484     16      Down 2.164181e-05 0.01784664 GO:0002484       BP
GO:0002486     15      Down 2.362061e-05 0.01784664 GO:0002486       BP
                                                                                                                        TERM
GO:0099010                                                                            modification of postsynaptic structure
GO:0060412                                                                                  ventricular septum morphogenesis
GO:0046703                                                                  natural killer cell lectin-like receptor binding
GO:0042832                                                                                     defense response to protozoan
GO:0099563                                                                                modification of synaptic structure
GO:0002483                                                 antigen processing and presentation of endogenous peptide antigen
GO:0019885                                 antigen processing and presentation of endogenous peptide antigen via MHC class I
GO:0002476                                antigen processing and presentation of endogenous peptide antigen via MHC class Ib
GO:0002484                  antigen processing and presentation of endogenous peptide antigen via MHC class I via ER pathway
GO:0002486 antigen processing and presentation of endogenous peptide antigen via MHC class I via ER pathway, TAP-independent</code></pre>
</div>
</div>
</section>
<section id="to-finish" class="level1">
<h1>To finish</h1>
<ul>
<li><p><em>Interpretations of sex-related variation are not always commensurate with the story the data actually tell</em> (Pape et al.&nbsp;2024). How do you judge the interpretations of the results in the published paper?</p></li>
<li><p>Discuss potential limitations of the paper and the conclusion about the absence of Tamoxifen effect.</p>
<ul>
<li>What is important to notice in the design of the experiment (tamoxifen treatment duration, batches, number of replicates, )?</li>
<li>Have a look at the results and materials and methods section: which analyses were made and how could they influence the results?</li>
</ul></li>
<li><p>Limitations of our reanalysis: what could be improved if we had more time?</p>
<ul>
<li>Try subsetting the dataset to one tissue as done by the authors. What do you notice?</li>
</ul></li>
<li><p><em>Sex is not a causal mechanism</em> (Pape et al.&nbsp;2024): follow-up on some DE genes and try to find which mechanism of action could explain their differential expression patterns.</p></li>
<li><p>Discuss what would be the next steps if that was your research project. Which experiments would you design next? How would you describe the results in a paper?</p>
<ul>
<li>You can also have a look at other related papers (e.g., https://www.nature.com/articles/s41586-022-04686-1, https://www.ahajournals.org/doi/10.1161/ATVBAHA.123.319922)</li>
<li>Or look at follow-up papers from the same authors (e.g., https://link.springer.com/article/10.1007/s12035-022-02860-0)</li>
</ul></li>
</ul>
<p><sub>Icons taken from http://www.flaticon.com/</sub></p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>